src/main/java/frc/robot/subsystems/ArmSystem.java

```diff
@@ -0,0 +1,58 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.*;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotMap;
+
+public class ArmSystem extends SubsystemBase {
+    private final CANSparkMax motor;
+    private final RelativeEncoder neoEncoder;
+    private final DutyCycleEncoder dutyCycleEncoder;
+
+    public ArmSystem() {
+        motor = new CANSparkMax(RobotMap.ARM_MOTOR_PORT, CANSparkLowLevel.MotorType.kBrushless);
+        neoEncoder = motor.getAlternateEncoder(RobotMap.ARM_ENCODER_CPR);
+        dutyCycleEncoder = new DutyCycleEncoder(RobotMap.ARM_ENCODER_PORT);
+    }
+
+    public void moveUp() {
+        if (getDutyCycleEncoderPosition() >= RobotMap.ARM_MAX_ANGLE){
```

Comment:
Because we aim to use the PIDF in the SparkMax for future parts, you should:

Use softLimits on the spark to limit its motion. These will only work with the NEO encoder and as such you must configure them according to the NEO encoder position values.  Read further [here](https://github.com/tomtzook/frc-learn-docs/blob/master/devices/rev/spark-max.md#limits). These limits will replace the checks for position you do. But test that these limits actually work well.

Reply:
done